### PR TITLE
Clippy 1.72 release fixes

### PR DESCRIPTION
**Pull request by Pat-Lafon**
With the new clippy release, I've gone through and either manually or `cargo clippy --fix`ed the triggered lints. I don't think anything here is particularly interesting/useful but may as well not let things slip.

### CODE REVIEW

File: lalrpop/src/lr1/build/test.rs
Comment (dburgener) [inline lalrpop/src/lr1/build/test.rs:284]: I vote against the comment that will get stale really quickly.

File: lalrpop/src/lr1/codegen/parse_table.rs
Comment (dburgener) [inline lalrpop/src/lr1/codegen/parse_table.rs:855]: I guess clippy made a suggestion on the generated code, motivating this change?

It looks like the logic is equivalent iff the series of indices is ordered and always increasing by one.  It should be ordered, since it's the result of `.enumerate()`, but can't it possibly not be increasing by one if the None case occurs on line 823 in between two instances of the Some() case?

File: lalrpop/src/lr1/build/test.rs
Comment (Pat-Lafon) [reply lalrpop/src/lr1/build/test.rs:284]: I'm confused? Yes it will get stale but the point is that, at that point, this allow can be removed. Clippy sometimes has false positives that get fixed in a subsequent version so we may as well keep track of false positives and when they can be removed.(Here the fix has already been merged into master)

File: lalrpop/src/lr1/codegen/parse_table.rs
Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Yes, clippy is more aggressively linting on or patterns of incrementally increasing values.

That's an interesting point, is there a small test case that helps trigger this? All of the current tests are of the logically equivalent form.

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Ok, nervermind, found one in lrgrammar.rs

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: I've attempted a fix for this to check the ordering first.

File: lalrpop/src/lr1/build/test.rs
Comment (dburgener) [reply lalrpop/src/lr1/build/test.rs:284]: Oh, I misunderstood.  I thought you were saying this allow wasn't needed yet, but would be needed after the clippy update to 1.72.  It's exactly the other way around.

In that case, maybe the comment would be more clear if it specified the expected action of removing this after 1.72?

File: lalrpop/src/lr1/codegen/parse_table.rs
Comment (dburgener) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: The fix looks right.  The weird case is exactly 1 terminal, which looks like it ends up using the inclusive range `N..=N`, instead of the old `N`.  That's probably a really uncommon case though, and an inclusive range of length one is just weird, not wrong.

Is this worth the extra code complexity?  Is it just about making clippy happy, or is there a performance benefit to replacing "1 | 2 | 3 | 4" with "1..=4"?

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: It compiles to the exact same code. Entirely just a clippy thing.

Comment (dburgener) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Well, a maintainer should have a final say, but I feel as though going from 1 fairly straightforward line to 13 lines with a fair amount of logic to parse and think though isn't worth it to make clippy happy in the generated code.  I would vote for just silencing this one to keep the lalrpop code more readable.

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Certainly up to the maintainer, I can understand the code is more complex than before to handle this case.

However, it's a little disingenuous to say it went from 1 line to 13 lines. For example, the old case is still there but now takes 5 lines because of formatting... that part didn't suddenly become more complex.

Comment (dburgener) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: I'd just call that "lines of code is a bad metric".  But it's a convenient shorthand.  Personally, I think the fact that you've got two cases instead of 1 is an increase in complexity, and the logic in the window closure, particularly with indexing is certainly more involved than a simple map.

It's not a huge difference, but the "make clippy happy" I would call a near zero benefit in this case, since clippy is really analyzing the readability of the generated code, but our level of care about that is near-zero, and our level of care about the readability of the actual lalrpop code is much higher.

Comment (youknowone) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: I agree to @dburgener. I believe clippy suggests it for readability.
But we maintain code generator, not the generated code here. People will read more generator code than generated code. In my opinion, previous one looks more readable.

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Fair, I've reverted this

Comment (youknowone) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: @Pat-Lafon will putting clippy ignore to line 808 work? 
https://github.com/lalrpop/lalrpop/pull/824/files#diff-f6fa48d7b0092fbf2bbdf4a7413478e304b5b2127fd314a3848b64e022f7aa71R808

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Sure, lets try that

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/parse_table.rs:855]: Marking this as resolved

**Review by youknowone — APPROVED** [review]
@Pat-Lafon @dburgener Thank you so much!

**Pull request merged**
